internal/io/fs.go

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2017-present SIGHUP s.r.l All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package io
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+func CheckDirIsEmpty(target string) error {
+	if _, err := os.Stat(target); os.IsNotExist(err) {
+		return nil
+	}
+
+	return filepath.Walk(target, func(path string, info os.FileInfo, err error) error {
+		return fmt.Errorf("the target directory is not empty: %s", path)
+	})
+}
+
+func AppendBufferToFile(b bytes.Buffer, target string) error {
+	destination, err := os.OpenFile(target, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	if err != nil {
+		return err
+	}
+
+	_, err = b.WriteTo(destination)
+	if err != nil {
+		return err
+	}
+
+	defer destination.Close()
```

Comment:
deferring here is of little value, either call it directly or (better) move it before `b.WriteTo(destination)`